### ISSUE TITLE
Pipeline Tool use same instance for same files

### ISSUE THREAD

**Issue by cra0zy** [mono/MonoGame#4658]
It would be nice to implement a check where:
 - check if 2 or more instances of Pipeline Tool are running
   - check if it has the same .mgcb file opened
     - bring instance to front

**Comment by tomspilman** [comment]
Yeah...  this is pretty simple to do.  The Pipeline tool uses a `EventWaitHandle` with a unique name based on the project file that is open.  On start we check if one already exists...  `EventWaitHandle.OpenExisting(name)`.  If it does exist we signal it and shut down the process.

Note that it should only do this on first launch of the pipeline tool when opening a .mgcb file via the command line.  We have to allow a user to open two pipeline instances and manually open the same project in both.

**Comment by aravol** [comment]
So should a manual open of a file ever "steal" the handle when it becomes available? If, for example, one opens a file, opens the same file manually, closes the first instance, and then tries to open the same file from the shell, should the second pipeline instance focus, or should the third instance take control of that handle?

**Comment by tomspilman** [comment]
> should the second pipeline instance focus, or should the third instance take control of that handle?

Good question...  I would think the manually opened instance should become the new main instance.  I think this should be easy to achieve.
